CIP-0164/README.md

```diff
@@ -0,0 +1,2774 @@
+  via the Mempool, but the adversary can prevent that for their EBs.) The
+  bitmap-based addressing scheme allows for compact requests for even thousands
+  of transactions: a few hundred bytes of MsgLeiosBlockTxsRequest can request
+  every transaction in even the largest EB, while a MsgLeiosBlockTxsRequest for
+  a single transaction would old cost tens of bytes. Without a compact
+  addressing scheme, a node that requires every transaction for some EB would
+  essentially need to send a request that's the same size as the EB itself,
+  which is large enough to be considered an unnecessary risk of increased
+  latency.
+- The client can request multiple votes at once, which avoids wasting resources
+  on overhead due to the hundreds of votes exchanged per EB. Because the first
+  vote in a bundle could have arrived sooner than the last vote in a bundle if
+  it had not been bundled, maximal bundling risks unnecessary increases in
+  latency. Some heuristic will balance the overhead decrease and latency
+  increase, such as the client gradually stopps bundling its vote requests as
+  its set of received votes approaches a quorum.
+- The server can do the same when bundling vote offers, but it should be more
+  conservative, in case the client is already closer to a quorum than it is.
+- MsgLeiosBlockRangeRequest lets syncing nodes avoid wasting resources on
+  overhead due to the (hopefully) high rate of EBs per RB. BlockFetch already
+  bundles its RB requests when syncing, and this message lets LeiosFetch do the
+  same. The starvation detection and avoidance mechanism used by Ouroboros
+  Genesis's Devoted BlockFetch variant can be easily copied for
+  MsgLeiosBlockRangeRequest if Ouroboros Genesis is enabled.
+- Recall that the `certified_eb` bit enables the client to correctly predict the
+  total payload size of the valid replies to a MsgLeiosBlockRangeRequest. This
+  enables the client to manage its receive buffers, balance load across peers,
+  etc.
+- A server should disconnect if the client requests an EB (or its transactions)
+  the server does not have. For young EBs, a client can avoid this by waiting
+  for MsgLeiosBlockOffer (or MsgLeiosBlockTxsOffer) before sending a request.
+  For old EBs, a ChainSync MsgRollForward for an RB that certifies its
+  predecessor's EB would also imply the server has selected that EB, and so must
+  be able to serve it. Whether additional restrictions would be useful is not
+  yet clear. For example, it seems natural to restrict MsgLeiosBlockRequest and
+  MsgLeiosBlockTxsRequest to young EBs (and perhaps also
+  MsgLeiosBlockRangeRequest to old EBs), but it's not already clear what the
+  benefit would be.
+- If MsgLeiosBlockRequest and MsgLeiosBlockTxsRequest were restricted to young
+  EBs, then MsgLeiosBlockRangeRequest would not only enable syncing nodes but
+  also the unfortunate node that suffers from a $\Delta^\text{A}_\text{EB}$
```

Comment:
This Delta^A might be a typo (Delta^W perhaps?), I do not see it defined elsewhere in the document.